### DISCUSSION TITLE
Open application with sign in account defined as parameter (start.cmd / Start-IntuneManagement.ps1) possible?

### DISCUSSION THREAD

**Discussion by ASMailer** [Micke-K/IntuneManagement#126] [General]
Hey Mike,

I have a question about the sign in process. Is there a possibility to open the "application/toolkit" with a pre defined sign in account, instead of selecting it inside the application menu on right side?

For example, pre define the account name in the "start.cmd", "Start-IntuneManagement.ps1" or in the "MSALAuthentication.psm1" ?

Thanks in advance for an hint
Greetings
Andi

**Comment by Micke-K** [comment]
Hello!

Not sure if I understand this correctly. The Pp with log you in with the last used user. Do you want to override that or is it something else?

Cheers!

**Reply by ASMailer** [reply]
We need to start the tool and to handover the used/target account "account@onmicrosoft.com" direct.
Instead of choose/select/start the account auth process, via the right corner icon.

![image](https://user-images.githubusercontent.com/79043883/231177187-8ce339f2-4c66-4c8c-87b9-ebd4104718df.png)

I am now looking for a suitable entry point where we could hand over, define fix, the account information.

Instead of start the auth process by clicking on the icon, get direct the MSAL form when open the tool.

thx
Andi

**Reply by ASMailer** [reply]
At the end, If I think about it, it would help to get direct the MSAL form when open the tool

**Reply by ASMailer** [reply]
Hi Mike, do you have any tip for me :-)

**Comment by Micke-K** [comment]
Row 985 in MSALAuthentication.psm1 should be

`if($global:MainAppStarted -and ((-not $authResult -and $Silent -ne $true) -or $prompConsent))`

Try changing that to:

`if(((-not $authResult -and $Silent -ne $true) -or $prompConsent))`

The app will still try to authenticate with last logged on user unless you untick "Remember login" in Settings.

What's the harm in clicking the login button? :)

Cheers!

**Reply by ASMailer** [reply]
Its about new security rules :-) ... , the login will be done in a isolated box/session remote, so we as admins doesn´t get (need) the pass furthermore 

I will try it out tomorrow 

Thank you very much Mike 
Andi

**Comment by Micke-K** [comment]
I just tested and it kind of "worked". I got a login popup but behind the splash screen and it was "empty". At least looks empty but there are previous logins I can click on when I hover the mouse over it. No clue if that is a bug in MSAL, something wrong in the code or something in my environment.

![image](https://user-images.githubusercontent.com/43226266/233099151-e38a78fa-e51f-45f9-8ba3-321c1ed5a344.png)

Cheers!

**Comment by ASMailer** [comment]
I has worked perfect once, and now I see the message in the log.

"Failed to login Exception: ActiveX control '8856f961-340a-11d0-a96b-00c04fd705a2' cannot be instantiated because the current thread is not in a single-threaded apartment."

... I found on the "stackoverflow" page an answer to the same ...

![image](https://user-images.githubusercontent.com/79043883/233179894-7df8f4ed-75f7-4e28-ba1f-e28450d662bf.png)
